rest-api/README.md

```diff
@@ -0,0 +1,174 @@
+                    "input" : ["start", "1", "0", "2", "2"]
+                }
+            },
+            {
+                "type" : "graph",
+                "id" : 1,
+                "model" : "graph {rankdir=TB;node [style=filled fillcolor=white shape=circle];4[];3[];2[];1[];4 -- 3 [];3 -- 1 [];3 -- 2 [label=a ];2 -- 1 [label=g ];}"
+            }
+        ]
+    }
+}
+```
+
+#### REQUEST - UPDATE INPUT
+This request updates the input for a specific model and reset the model to its initial state. Note: Only valid for models with input.
+
+```js
+url: BASE_URL + "/init/input",
+method: "POST",
+headers: {"Content-Type": "application/json"},
+body: {
+  "id": {model-id}, // Might be better to have the id as a parameter in the url?
+  "input" : {new-input}
+}
+```
+
+###### Sample request body
+```js
+body: {
+  "id": 0,
+  "input" : "1000"
+}
+```
+
+###### Sample output
+```json
+{
+    "data" : {
+        "type" : "nfa",
+        "id" : 0,
+        "model" : "digraph {rankdir=LR;node [style=filled fillcolor=white shape=circle];start[fontcolor=white color=white];a[shape=doublecircle];b[  ];c[  ];d[  ];e[  ];f[  ];start -> a [label=start style=solid];a -> b [label=1 ];b -> c [label=0 ];c -> d [label=2 ];c -> e [label=2 ];d -> a [label=1 ];e -> f [label=1 ];}",
+        "simulation" : {
+            "status" : "OK",
+            "current-input-index" : 0,
+            "input" : ["start", "1", "0", "0", "0"]
+        }
+    }
+}
+```
+
+## TRANSITION
+The requests under this category are responsible for changing the state of a specific model. The state transition function called when handling the request is specialized and specific for each model.
+Ex: In the case of a DFA/NFA, the next state refers to the next configuration.
+
+#### REQUEST - NEXT STATE
+
+<code>GET: BASE_URL + "/transition/next?id={model-id}"</code>  
+
+#### REQUEST - PREVIOUS STATE
+
+<code>GET: BASE_URL + "/transition/previous?id={model-id}"</code>
+
+###### Sample request
+<code>GET: BASE_URL + "/transition/next?id=0"</code>
+
+###### Sample output
+```json
+{
+    "data" : {
+        "type" : "nfa",
+        "id" : 0,
+        "model" : "digraph {rankdir=LR;node [style=filled fillcolor=white shape=circle];start[fontcolor=white color=white];a[shape=doublecircle];b[fillcolor=darkgreen fontcolor = white];c[  ];d[  ];e[  ];f[  ];start -> a [label=start ];a -> b [label=1 color=darkgreen style=bold];b -> c [label=0 color=black style=dashed];c -> d [label=2 color=black style=dashed];c -> e [label=2 color=black style=dashed];d -> a [label=1 color=black style=dashed];e -> f [label=1 color=black style=dashed];}",
+        "simulation" : {
+            "status" : "OK",
+            "current-input-index" : 1,
+            "input" : ["start", "1", "0", "0", "0"]
+        }
+    }
+}
+```
+###### Visualized model before the request
+<img src="./state0.jpg" width="300px"/>
+
+###### Visualized model after the request
+<img src="./state1.jpg" width="300px"/>
+
+## OUTPUT
```

Comment:
I am not sure I understand this. Why is this called output? When I look at the example, it looks like it is a feedback when setting some parameter. In an electrical circuit, it would be e.g., changing the resistance (in the GUI) and then re-run the simulation.